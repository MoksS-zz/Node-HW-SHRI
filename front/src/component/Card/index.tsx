import React, { FC } from "react";
import Date from "../Date";
import { getDate, getDuration } from "../../helpers/time";
import "./Card.scss";

export interface CardProps {
  start?: string;
  duration?: number;
  column?: boolean;
  status?: string;
  number?: number;
  commit?: string;
  branch?: string;
  hash?: string;
  author?: string;
};

const Card: FC<CardProps> = (props) => {
  const date = props.start ? getDate(props.start) : false;

  const duration = getDuration(props.duration!);

  const status = `Icon Icon__indentRigth-s Icon__size-default Icon__img-${props.status}`

  return (
    <div className={props.column ? "Card Card__indentBottom-s Card__column-on" : "Card Card__indentBottom-s"}>
      <div className="Card_Info Card__flexBasic-80">
        <div className="Card_Build Card_Build__indentBottom-s">
          <div className="Card_Status">
            <span className={status}></span>
            <span className="Text Text__size-l Text__weight-normal Text__lineHeight-xl Text__color-green">#{props.number}</span>
          </div>
          <span className="Text Text__indentLeft-xs Text__size-s Text__weight-small Text__lineHeight-xl Text__color-default">{props.commit}</span>
        </div>
        <div className="Card_Commit Card_Commit__indentLeft-xxl">
          <div className="Card_Branch Card_Branch__indentRigth-l ">
            <span className="Icon Icon__indentRigth-xs Icon__size-xs Icon__img-code"></span>
            <span className="Text Text__indentRigth-xs Text__size-m Text__weight-small Text__lineHeight-l Text__color-default">{props.branch}</span>
            <span className="Text Text__indentRigth-xs Text__size-m Text__weight-small Text__lineHeight-l Text__color-hash">{props.hash}</span>
          </div>
          <div className="Card_User">
            <span className="Icon Icon__indentRigth-xs Icon__size-xs Icon__img-user"></span>
            <span className="Text Text__size-m Text__weight-small Text__lineHeight-l Text__color-default">{props.author}</span>
          </div>
        </div>
      </div>
      {date === false ? 
        "" :
        <Date
        startDate={date}
        duration={duration}
      />
      }
    </div>

  );
}

export default Card;