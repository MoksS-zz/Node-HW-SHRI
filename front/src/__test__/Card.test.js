import React from "react";
import renderer from "react-test-renderer";
import Card from "../component/Card";

// jest.mock('react-router-dom', () => ({
//   useHistory: jest.fn()
// }))

it("renders Card correctly", () => {
  const logo = renderer
    .create(<Card 
      start={"12-09-1997 12:45"}
      duration={123422}
      status={"Success"}
      number={12}
      commit={"КОммит Коммит АГАСЬ"}
      branch={"master"}
      hash={"asd213sf213a1as2"}
      author={"MoksS"}
    />)
    .toJSON();

  expect(logo).toMatchSnapshot();
});
