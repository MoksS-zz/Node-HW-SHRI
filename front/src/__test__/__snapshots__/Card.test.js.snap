// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card correctly 1`] = `
<div
  className="Card Card__indentBottom-s"
>
  <div
    className="Card_Info Card__flexBasic-80"
  >
    <div
      className="Card_Build Card_Build__indentBottom-s"
    >
      <div
        className="Card_Status"
      >
        <span
          className="Icon Icon__indentRigth-s Icon__size-default Icon__img-Success"
        />
        <span
          className="Text Text__size-l Text__weight-normal Text__lineHeight-xl Text__color-green"
        >
          #
          12
        </span>
      </div>
      <span
        className="Text Text__indentLeft-xs Text__size-s Text__weight-small Text__lineHeight-xl Text__color-default"
      >
        КОммит Коммит АГАСЬ
      </span>
    </div>
    <div
      className="Card_Commit Card_Commit__indentLeft-xxl"
    >
      <div
        className="Card_Branch Card_Branch__indentRigth-l "
      >
        <span
          className="Icon Icon__indentRigth-xs Icon__size-xs Icon__img-code"
        />
        <span
          className="Text Text__indentRigth-xs Text__size-m Text__weight-small Text__lineHeight-l Text__color-default"
        >
          master
        </span>
        <span
          className="Text Text__indentRigth-xs Text__size-m Text__weight-small Text__lineHeight-l Text__color-hash"
        >
          asd213sf213a1as2
        </span>
      </div>
      <div
        className="Card_User"
      >
        <span
          className="Icon Icon__indentRigth-xs Icon__size-xs Icon__img-user"
        />
        <span
          className="Text Text__size-m Text__weight-small Text__lineHeight-l Text__color-default"
        >
          MoksS
        </span>
      </div>
    </div>
  </div>
  <div
    className="Date"
  >
    <div
      className="Date_Day"
    >
      <span
        className="Icon Icon__indentRigth-xs Icon__size-xs Icon__img-calendar"
      />
      <span
        className="Text Text__size-m Text__weight-small Text__lineHeight-l Text__color-secondary Text__indentRigth-m Date__DinamicColor-on"
      >
        9 дек, 12:45
      </span>
    </div>
    <div
      className="Date_Hour"
    >
      <span
        className="Icon Icon__indentRigth-xs Icon__size-xs Icon__img-clock"
      />
      <span
        className="Text Text__size-m Text__weight-small Text__lineHeight-l Text__color-secondary Date__DinamicColor-on"
      >
        2 мин 3сек
      </span>
    </div>
  </div>
</div>
`;
